examples/test_PopEllSliceSampler.py

```diff
@@ -0,0 +1,140 @@
+
+    if args.funnel:
+        sigma = args.sigma
+        centers = np.sin(np.arange(ndim) / 2.)
+        data = np.random.normal(centers, sigma).reshape((1, -1))
+
+        def loglike(theta):
+            sigma = 10**theta[:,0]
+            
+            like = -0.5 * (((theta[:,1:] - data)/sigma.reshape((-1, 1)))**2).sum(axis=1) - 0.5 * np.log(2 * np.pi * sigma**2) * ndim
+            return like
+
+        def transform(x):
+            z = x * 20 - 10
+            z[:,0] = x[:,0] * 6 - 3
+            return z
+        import string
+        #print(ndim, len(list(string.ascii_lowercase)))
+        paramnames = ['sigma'] + ['param%d' % (i+1) for i in range(ndim)][:ndim]
+
+    
+    from ultranest import ReactiveNestedSampler
+    sampler = ReactiveNestedSampler(paramnames, loglike,\
+    transform=transform, log_dir=args.log_dir, resume='overwrite',\
+    draw_multiple=False, vectorized=True,)
+    if args.ElliSlice:
+        import ultranest.popstepsampler as ultrapop
+        direction=[ultrapop.generate_cube_oriented_direction,ultrapop.generate_random_direction, ultrapop.generate_region_oriented_direction, ultrapop.generate_region_random_direction]
+        sampler.stepsampler = ultrapop.PopulationEllipticalSliceSampler(popsize=args.popsize,nsteps=args.nstep,generate_direction=direction[1],scale=1.0,scale_adapt_factor=0.9)
+    if args.PopSlice:
+        import ultranest.popstepsampler as ultrapop
+        direction=[ultrapop.generate_cube_oriented_direction,ultrapop.generate_random_direction, ultrapop.generate_region_oriented_direction, ultrapop.generate_region_random_direction]
+        sampler.stepsampler = ultrapop.PopulationSliceSampler(popsize=args.popsize,nsteps=args.nstep,generate_direction=direction[1],scale=1.0)
+    if args.Slice:
+        import ultranest.stepsampler as stepsampler
+        sampler.stepsampler = stepsampler.SliceSampler(nsteps=args.nstep,generate_direction=stepsampler.generate_mixture_random_direction,)
+    if args.PopGaussWalk:
+        import ultranest.popstepsampler as ultrapop
+        direction=[ultrapop.generate_cube_oriented_direction,ultrapop.generate_random_direction, ultrapop.generate_region_oriented_direction, ultrapop.generate_region_random_direction]
+        sampler.stepsampler = ultrapop.PopulationRandomWalkSampler(popsize=args.popsize, nsteps=args.nstep, generate_direction=direction[1],scale=1.0,)
+
+    sampler.run(frac_remain=0.5, min_num_live_points=args.num_live_points, max_num_improvement_loops=1)
+    sampler.print_results()
+    if ndim <= 20:
+        sampler.plot()
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument('--x_dim', type=int, default=2,
+                        help="Dimensionality")
+    parser.add_argument("--num_live_points", type=int, default=400)
+    parser.add_argument('--log_dir', type=str)
+    parser.add_argument('--seed', type=int, default=0)
+    parser.add_argument('--rosenbrock', action='store_true')
```

Comment:
This is really nitpicking, so feel free to ignore, but you could use --problem=multishell , i.e, argparse options